File: ABAC/frontend/src/app/models/planet.ts
import { PlanetStatus } from './status';
import Team from './team';

export default class Planet {
  public id: number;
  public name: string;
  public description: string;
  public image: string;
  public planetStatus: PlanetStatus;
  public team: Team;

  constructor(
    id: number,
    name: string,
    description: string,
    image: string,
    status: PlanetStatus,
    team: Team
  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.image = `./assets/${image}`;
    this.planetStatus = status;
    this.team = team;
  }
}

File: ABAC/backend/backend/Entities/Planet.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Entities
{
    public class Planet
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        public string Description { get; set; } = String.Empty;
        public string Image { get; set; }
        public PlanetStatus PlanetStatus { get; set; } = PlanetStatus.TODO;
        public int TeamId { get; set; }
        public Team Team { get; set; }
    }
}
